next/src/component/link/data-access-link/useGetLinks.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useEffect } from "react";
+import { axiosInstance } from "component/sharing/util";
+import { mapLinksData } from "component/link/util-map/mapLinksData";
+import { useAsync } from "component/sharing/util";
+import { ALL_LINKS_ID } from "./constant";
+
+interface MapDataFormatProps {
+  id: string | number;
+  created_at: string | number;
+  url: string;
+  image_source: string;
+  title: string;
+  description: string;
+}
+interface AsyncProps {
+  execute: any;
+  loading: boolean;
+  data: any;
+}
```

Comment:
data에 대한 타입은 여러가지가 올 수 있으니까 이 부분을 제너릭 타입으로 쓸 수 있겠죠?
`execute`함수의 타입은 F로 `data` 변수의 타입은 T로 사용하여 사용 할 수 있습니다. 
어떻게하면 좋을지 한번 고민해보세요 ! 

```suggestion
interface AsyncProps<T = any,F=any> {
    execute: F;
    loading: boolean;
    data: T;
}```